### ISSUE TITLE
debug/log flag

### ISSUE THREAD

**Issue by mrexodia** [keystone-engine/keystone#206]
Just a quick question, do you think we should add a flag to keystone that will log all `assemble` attempts and their results? This could help for detecting inconsistencies, however it might be out of the scope for the project.

**Comment by aquynh** [comment]
what do you mean by "log all assemble attempts and their results"?

**Comment by mrexodia** [comment]
Well as it says. Log all calls to the api (instruction text +result codes
and data).

**Comment by aquynh** [comment]
why does it belong to this low level framework?

**Comment by mrexodia** [comment]
I do believe so yes. Obviously this would only be included when enabled with a compile option.

The reason was that me and @fvrmatteo were discussing bugs in keystone and a log would have really helped with reproducing the issues.

If you don't want to include it, makes sense but maybe we can provide a simple wrapper header that has this so people can provide these dumps in an organized way.

**Comment by aquynh** [comment]
This kind of stuff should be put under suite/ directory. Pull req is welcome.

**Comment by mrexodia** [comment]
Yeah I might take a look at it later and write some code. I'm very busy with x64dbg at the moment though so that's why I opened an issue.
